### ISSUE TITLE
Examples containing special characters leads to ValueError: unknown training file format : unk for file ./data/nlu.md

### ISSUE THREAD

**Issue by jganseman** [RasaHQ/rasa_nlu#832]
**Rasa NLU version** (e.g. `0.7.3`): 0.11.1

**Used backend / pipeline** (`mitie`, `spacy_sklearn`, ...): spacy_sklearn

**Operating system** (windows, osx, ...): Windows 7 Enterprise using Anaconda

**Issue**: If the file **./data/nlu.md** contains example sentences that include special characters, such as the cedille in 
```
## intent:french
- Do you speak French?
- Parlez-vous français
```
then training fails with the following stacktrace:
```
C:\Users\jg\>python -m rasa_nlu.train -c nlu_model_config.json --fixed_model_name current
INFO:rasa_nlu.utils.spacy_utils:Trying to load spacy model with name 'nl'
INFO:rasa_nlu.components:Added 'nlp_spacy' to component cache. Key 'nlp_spacy-nl'.
INFO:rasa_nlu.converters:Training data format at ./data/nlu.md is unk
Traceback (most recent call last):
  File "C:\Users\jg\AppData\Local\Continuum\anaconda3\envs\rasatests\lib\runpy.py", line 193, in _run_module_as_main "__main__", mod_spec)
  File "C:\Users\jg\AppData\Local\Continuum\anaconda3\envs\rasatests\lib\runpy.py", line 85, in _run_code exec(code, run_globals)
  File "C:\Users\jg\AppData\Local\Continuum\anaconda3\envs\rasatests\lib\site-packages\rasa_nlu\train.py", line 125, in <module> do_train(config)
  File "C:\Users\jg\AppData\Local\Continuum\anaconda3\envs\rasatests\lib\site-packages\rasa_nlu\train.py", line 113, in do_train training_data = load_data(config['data'], config['language'])
  File "C:\Users\jg\AppData\Local\Continuum\anaconda3\envs\rasatests\lib\site-packages\rasa_nlu\converters.py", line 370, in load_data "file {}".format(fformat, resource_name))
ValueError: unknown training file format : unk for file ./data/nlu.md
```
( Problem exists for both native and UTF-8 file encodings, and also when using default "en" as language )

**Content of configuration file** (if used & relevant):
```json
{
  "pipeline": "spacy_sklearn",
  "path": "./models/nlu",
  "data": "./data/nlu.md",
  "language": "nl"
}
```

**Comment by jganseman** [comment]
This may possibly be related to bug #719

**Comment by tmbo** [comment]
@jganseman which python version is that?

**Comment by jganseman** [comment]
Python 3.6.4 :: Anaconda, Inc.

**Comment by tmbo** [comment]
Alright, let me try to reproduce that.

**Comment by tmbo** [comment]
I couldn't reproduce the issue with python 3 on my local machine and that data. I used the latest master version though, do you mind installing that and trying it out (we have done some refactoring on the data loading so that might have fixed / changed the behaviour).

**Comment by jganseman** [comment]
I hope to look at it in more detail this weekend! Also running into some troubles with strings that contain escaped characters now, but indeed, could be an issue of outdated or conflicting packages - conda and pip seem not always to work well together.

**Comment by jganseman** [comment]
I've tried reproducing on a Lubuntu VM with the same version 0.11.1 force-installed through pip, but everything runs just fine there.

It apparently has to do with the `unk` in `INFO:rasa_nlu.converters:Training data format at ./data/nlu.md is unk`
which in Linux is just: 
`INFO:rasa_nlu.converters:Training data format at ./data/nlu.md is md` . 

I'll re-check the windows files for anything peculiar regarding encoding, or it could be something to do with windows-specific file parsing.

**Comment by jganseman** [comment]
Update: the problem is probably related to other characters than ç (error messages of the yaml parser aren't too informative so took a while to figure out)
- On linux, if domain.yml contains escaped characters in the templates, e.g.
```
  utter_goodbye:
  - text: "Bye ! Visit http://example.com\#12345 for more info."
```
 I get the following error:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/train.py", line 136, in <module>
    additional_arguments)
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/train.py", line 94, in train_dialogue_model
    agent = Agent(domain_file, policies=[MemoizationPolicy(), KerasPolicy()])
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/agent.py", line 42, in __init__
    self.domain = self._create_domain(domain)
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/agent.py", line 237, in _create_domain
    return TemplateDomain.load(domain)
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/domain.py", line 429, in load
    cls.validate_domain_yaml(filename)
  File "/home/joachim/.local/lib/python2.7/site-packages/rasa_core/domain.py", line 463, in validate_domain_yaml
    schema_files=[schema_file])
  File "/home/joachim/.local/lib/python2.7/site-packages/pykwalify/core.py", line 78, in __init__
    raise CoreError(u"Unable to load any data from source yaml file")
pykwalify.errors.CoreError: <CoreError: error code 3: Unable to load any data from source yaml file: Path: '/'>
```
On Windows, the above error appears, and there seem to be more characters that are problematic - including Unicode quotes “ ” (U+201C, U+201D).

May be related to encoding detection routine in the file reader?

**Comment by tmbo** [comment]
@jganseman I think your last messages is more related to rasa core.

Concerning the one before that, can you try using the latest master? There have been a number of changes to the data loading and the used encoding.

**Comment by robwebbjr** [comment]
Hi.
Here's the answer (at least it worked for me):

From http://pyradiomics.readthedocs.io/en/latest/faq.html

"Error loading parameter file
When I try to load my own parameter file, I get error:”CoreError: Unable to load any data from source yaml file”
This error is thrown by PyKwalify when it is unable to read the parameter file. The most common cause is when the file is indented using tabs, which throws a “token (‘t’) not recognized error”. Instead, ensure that all indentations are made using 2 or 4 spaces."

I thought I was being a good student by typing the code, but I used tabs; I went back and did a copy-paste from the tutorial, and problem solved!

Robert
